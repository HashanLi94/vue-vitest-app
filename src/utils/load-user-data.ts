interface User {
	id: number;
	username: string;
	name: string;
	projects: string[];
	coolness?: number;
	favoriteFood: string;
}

async function fetchUser(username: string): Promise<User | undefined> {
	const users: User[] = [
		{
			id: 1,
			username: "evanYou",
			name: "Evan You",
			projects: ["vue.js", "nuxt", "vite"],
            favoriteFood: "Sushi"
		},
		{
			id: 2,
			username: "anthonyFu07",
			name: "Anthony Fu",
			projects: ["vitest", "vite"],
            favoriteFood: "Sushi"
		},
		{
			id: 3,
			username: "hashandu94",
			name: "Hashan Liyanage",
			projects: ["CCF", "SLA", "PP3"],
            favoriteFood: "Biriyani"
		},
	];

	return users.find((user) => user.username === username);
}

export async function getUserByUsername(username: string) {
	const user = await fetchUser(username);

	if (user) {
		user.coolness = username === "evenYou" ? 100 : -1;
	} else {
		throw new Error("User not found");
	}

	return user;
}
