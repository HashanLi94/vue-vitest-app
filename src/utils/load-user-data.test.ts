import { describe, test, expect } from "vitest";
import { getUserByUsername } from "./load-user-data";

describe("getUserByUsername", () => {
	test("fetch user", async () => {
		const user = await getUserByUsername("hashandu94");

		// expect(user).toEqual({
		// 	coolness: -1,
		// 	id: 3,
		// 	name: "Hashan Liyanage",
		// 	projects: ["CCF", "SLA", "PP3"],
		// 	username: "hashandu94",
		// });

		expect(user).toMatchSnapshot();
	});
});

describe("checkCoolness", () => {
	test("fetch user and check coolness", async () => {
		const userOne = await getUserByUsername("hashandu94");

		expect(userOne.coolness).toEqual(-1);
	});
});

describe("check the invalid user", () => {
	// test.fails("fetch user and check the user is valid or not", async () => {
	// 	const invalidUser = await getUserByUsername("hashandu948");
	// });

	test("fetch user and check the user is valid or not", async () => {
		expect(
			async () => await getUserByUsername("hashandu948")
		).rejects.toThrowError("User not found");
	});
});
