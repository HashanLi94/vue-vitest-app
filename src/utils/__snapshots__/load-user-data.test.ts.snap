// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getUserByUsername > fetch user 1`] = `
{
  "coolness": -1,
  "favoriteFood": "Biriyani",
  "id": 3,
  "name": "Hashan Liyanage",
  "projects": [
    "CCF",
    "SLA",
    "PP3",
  ],
  "username": "hashandu94",
}
`;
